src/redisearch_api.c

```diff
@@ -437,15 +437,17 @@ static void RS_IndexOptionsSetFlags(RSIndexOptions* options, uint32_t flags) {
   options->flags = flags;
 }
 
-#define REGISTER_API(name)                                                   \
-  if (moduleRegisterApi("RediSearch_" #name, RS_##name) != REDISMODULE_OK) { \
-    printf("could not register RediSearch_" #name "\r\n");                   \
-    return REDISMODULE_ERR;                                                  \
+#define REGISTER_API(name)                                                                  \
+  if (RedisModule_ExportSharedAPI(ctx, "RediSearch_" #name, RS_##name) != REDISMODULE_OK) { \
+    RedisModule_Log(ctx, "warning", "could not register RediSearch_" #name "\r\n");         \
+    return REDISMODULE_ERR;                                                                 \
   }
 
-int moduleRegisterApi(const char* funcname, void* funcptr);
-
 int RS_InitializeLibrary(RedisModuleCtx* ctx) {
+  if (RedisModule_ExportSharedAPI == NULL) {
```

Comment:
this will not allow to use redisearch on versions that does not supply this api (RedisModule_ExportSharedAPI). We should just print a warning. But not return an error.

Reply:
module.c will not fail if this function returns an error